crates/core/tedge_api/src/device_profile.rs

```diff
@@ -39,20 +39,28 @@ impl CommandPayload for DeviceProfileCmdPayload {
 #[derive(Debug, Deserialize, Serialize, Eq, PartialEq, Clone)]
 #[serde(rename_all = "camelCase")]
 pub struct DeviceProfileOperation {
-    operation: OperationType,
-    skip: bool,
     #[serde(flatten)]
-    payload: OperationPayload,
+    pub operation: Operation,
+    pub skip: bool,
+}
+
+#[derive(Debug, Deserialize, Serialize, Eq, PartialEq, Clone)]
+#[serde(untagged)]
+// If an operation has a payload, it should match WithPayload
+#[serde(deny_unknown_fields)]
+pub enum Operation {
+    WithPayload(OperationPayload),
+    JustType { operation: OperationType },
```

Comment:
I didn't fully understand the purpose of that `JustType` variant with only the `OperationType`. A device profile payload with only an `operation` field, without an associated `payload` field must be invalid. Right now the following payload (without a `payload` field for the `operation`) is accepted:
```json
{
  "status": "init",
  "name": "dev-profile",
  "version": "v2",
  "operations": [
    {
      "operation": "firmware_update",
      "@skip": false
    }
  ]
}
```

Reply:
right, tuple struct will be sufficient here.